### ISSUE TITLE
Delay in Cells iterator when worksheet contains validation rules

### ISSUE THREAD

**Issue by tjma1264** [ClosedXML/ClosedXML#1585]
## Read and complete the full issue template

Do not randomly delete sections. They are here for a reason.

**Do you want to request a *feature* or report a *bug*?**
- [x] Bug
- [ ] Feature
- [ ] Question

**Did you test against the latest CI build?**
- [x] Yes
- [ ] No

If you answered `No`, please test with [the latest development build](https://ci.appveyor.com/project/ClosedXML/ClosedXML/branch/develop/artifacts) first.

**Version of ClosedXML**

0.95.4

**What is the current behavior?**

Iterating cells in a row can take seconds to complete when worksheet contains validation rules.

**What is the expected behavior or new feature?**

Iterating cells in a row to take a fraction of a second.

**Is this a regression from the previous version?**

Yes.

No delay see when using v0.93.1.  Problem first appears in v0.94.0.

## Reproducibility
**This is an important section. Read it carefully. Failure to do so will cause a 'RTFM' comment.**

Without a code sample, it is unlikely that your issue will get attention. Don't be lazy. Do the effort and assist the developers to reproduce your problem. Code samples should be [minimal complete and verifiable](https://stackoverflow.com/help/mcve). Sample spreadsheets should be attached whenever applicable. Remove sensitive information.

**Code to reproduce problem:**
```c#
public void Main()
{
	var fileReader = File.OpenRead("test.xlsx");
	var workbook = new XLWorkbook(fileReader);
	var worksheet = workbook.Worksheet("Sheet1");

        foreach (var row in worksheet.Rows())
	{
               Console.WriteLine(row.Cells().ToList().Count());
	}
}
```

- [x] I attached a sample spreadsheet.  (You can drag files on to this issue)
[test.xlsx](https://github.com/ClosedXML/ClosedXML/files/5740035/test.xlsx)

**Comment by AyooshAgarwal** [comment]
I am also facing this issue, it's a serious one because the delay is caused by a sudden surge of memory consumption.
In my case a simple excel file < 10 records 13Kb in size with 10 columns (having a single column table with validation list) caused an increase in RAM consumption by around 4.5 GB

After some inspection, I found that the culprit is in the 

ClosedXML -> Excel -> XLWorkbook_Load.cs -> LoadSpreadsheetDocument() 
 #region LoadTables

Starting from Line 308

If the code inside this region (#region LoadTables) is commented, the issue dissappears.

Requesting the developers to kindly look into this asap, because this huge memory consumption can cripple production servers.

Regards,
Ayoosh

**Comment by vbjay** [comment]
Note the ...Used functions. Only loop through used rows or columns.  You
are looping through all rows and getting all cells.

**Comment by AyooshAgarwal** [comment]
Yeah, I understand, but is it going through the all the blank rows too ,incase of data tables (which are unused from user's perspective) ?
Could you try it with this sample file ?
[sample.xlsx](https://github.com/ClosedXML/ClosedXML/files/6083299/sample.xlsx)

**Comment by AyooshAgarwal** [comment]
Any Updates ???
